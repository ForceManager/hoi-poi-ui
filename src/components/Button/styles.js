export default (theme) => ({
    root: {
        position: 'relative',
        boxSizing: 'border-box',
        display: 'inline-block',
        minWidth: '100px',
        padding: '5px 16px',
        borderRadius: '100px',
        textAlign: 'center',
        cursor: 'pointer !important',
        '&:active': {
            ...theme.effects.buttonActive,
        },
        '&$fullWidth': {
            width: '100%',
        },
        '&$disabled': {
            backgroundPosition: '0 0 !important',
            opacity: 0.5,
            cursor: 'not-allowed !important',
            '&:hover': {
                backgroundPosition: '0 0 !important',
            },
            '& *': {
                cursor: 'not-allowed',
            },
        },
        '& *': {
            cursor: 'pointer',
        },
    },
    label: {
        isolate: true,
        display: 'inherit',
        alignItems: 'inherit',
        justifyContent: 'inherit',
        fontWeight: 500,
        transition: 'opacity .2s ease-in-out',
        ...theme.typography.button,
    },
    labelLoading: {
        isolate: true,
        position: 'relative',
        '& $label': {
            opacity: 0,
        },
        '& $loader': {
            opacity: 1,
        },
    },
    loader: {
        isolate: true,
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        opacity: 0,
        transition: 'opacity .2s ease-in-out',
    },
    // Types
    outlined: {
        padding: '4px 16px',
        '&$primary': {
            background: 'transparent',
            backgroundColor: 'transparent',
            border: `1px solid ${theme.colors.primary}`,
            transition: `color .2s ease-in-out, box-shadow .2s ease-in-out, background-color .1s ease-in-out`,
            '&$disabled': {
                '&:hover': {
                    backgroundColor: 'transparent',
                    '& $label': {
                        color: theme.colors.primary,
                    },
                },
            },
            '& $label': {
                color: theme.colors.primary,
            },
            '&:hover': {
                backgroundColor: theme.colors.primary,
                '& $label': {
                    color: theme.colors.text.primary,
                },
            },
        },
        '&$danger': {
            background: 'transparent',
            backgroundColor: 'transparent',
            border: `1px solid ${theme.colors.danger}`,
            transition: `color .2s ease-in-out, box-shadow .2s ease-in-out, background-color .1s ease-in-out`,
            '& $label': {
                color: theme.colors.danger,
            },
            '&:hover': {
                backgroundColor: theme.colors.danger,
                '& $label': {
                    color: theme.colors.text.danger,
                },
            },
        },
        '&$small': {
            padding: '0 16px',
        },
        '&$big': {
            padding: '9px 16px',
        },
    },
    squared: {
        borderRadius: 2,
        padding: '4px 9px',
        minWidth: '0',
        background: 'transparent',
        backgroundColor: 'transparent',
        border: `1px solid ${theme.colors.placeholders}`,
        transition: `border-color .2s ease-in-out`,
        '&:hover': {
            border: `1px solid ${theme.colors.placeholdersHover}`,
            background: 'transparent',
            backgroundColor: 'transparent',
        },
        '&$disabled': {
            '&:hover': {
                border: `1px solid ${theme.colors.placeholders}`,
                backgroundColor: 'transparent',
            },
        },
        '&$white': {
            background: 'transparent',
            backgroundColor: 'transparent',
            '&$disabled': {
                '&:hover': {
                    backgroundColor: 'transparent',
                },
            },
            '&:hover': {
                background: 'transparent',
                backgroundColor: 'transparent',
            },
        },
        '&$primary': {
            background: 'transparent',
            backgroundColor: 'transparent',
            '&$disabled': {
                '&:hover': {
                    backgroundColor: 'transparent',
                    '& $label': {
                        color: theme.colors.primary,
                    },
                },
            },
            '& $label': {
                color: theme.colors.primary,
            },
        },
        '&$danger': {
            background: 'transparent',
            backgroundColor: 'transparent',
            '& $label': {
                color: theme.colors.danger,
            },
        },
        '&$small': {
            padding: '0 9px',
            minWidth: '0',
        },
        '&$big': {
            padding: '9px 9px',
            minWidth: '0',
        },
    },
    //Colors
    white: {
        backgroundColor: 'transparent',
        transition: `box-shadow .2s ease-in, background-color .1s ease-in-out`,
        '& $label': {
            color: theme.colors.text.default,
        },
        '&:hover': {
            backgroundColor: theme.colors.lightGray,
        },
    },
    primary: {
        background: `linear-gradient(to bottom, ${theme.colors.primary} 0%, ${
            theme.colors.primary
        } 50%, ${theme.colors.primaryDark} 100%)`,
        filter: `DXImageTransform.Microsoft.gradient( startColorstr='${
            theme.colors.primary
        }', endColorstr='${theme.colors.primaryDark}', GradientType=0 )`,
        backgroundSize: 'auto 200% !important',
        backgroundPosition: '100% 0 !important',
        transition: `box-shadow .2s ease-in, background-position .1s ease-in-out`,
        '& $label': {
            color: theme.colors.text.primary,
        },
        '&:hover': {
            backgroundPosition: '0 100% !important',
        },
    },
    danger: {
        background: `linear-gradient(to bottom, ${theme.colors.danger} 0%, ${
            theme.colors.danger
        } 50%, ${theme.colors.dangerDark} 100%)`,
        filter: `DXImageTransform.Microsoft.gradient( startColorstr='${
            theme.colors.danger
        }', endColorstr='${theme.colors.dangerDark}', GradientType=0 )`,
        backgroundSize: 'auto 200% !important',
        backgroundPosition: '100% 0 !important',
        transition: `box-shadow .2s ease-in, background-position .1s ease-in-out`,
        '& $label': {
            color: theme.colors.text.danger,
        },
        '&:hover': {
            backgroundPosition: '0 100% !important',
        },
    },
    // Sizes
    small: {
        minWidth: '60px',
        padding: '0px 10px',
    },
    big: {
        minWidth: '130px',
        padding: '10px 16px',
    },
    // States
    disabled: {},
    fullWidth: {},
});
