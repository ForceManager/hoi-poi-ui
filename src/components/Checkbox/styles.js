export default (theme) => ({
    root: {
        cursor: 'pointer',
    },
    input: {
        display: 'none',
    },
    disabled: {
        opacity: 0.5,
        cursor: 'not-allowed',
    },
});
