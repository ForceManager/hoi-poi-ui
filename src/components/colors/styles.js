export default (theme) => ({
    root: {},
    section: {
        paddingBottom: 20,
        marginBottom: 20,
        borderBottom: `1px solid ${theme.colors.grey[100]}`,
        '&:last-child': {
            borderBottom: 'none',
            marginBottom: 0,
        },
    },
    sectionTitle: {
        color: theme.colors.textLight.secondary,
        textTransform: 'capitalize',
        marginBottom: 12,
    },
    swatches: {
        display: 'flex',
        flexFlow: 'row wrap',
        padding: 12,
        gap: 12,
    },
    swatchesDark: {
        backgroundColor: theme.colors.background.black,
    },
    swatch: {
        display: 'flex',
        flexFlow: 'column nowrap',
        justifyContent: 'flex-end',
        alignItems: 'flex-end',
        width: 120,
        height: 60,
        padding: 8,
    },
    swatchTextDark: {
        color: theme.colors.primary.black,
    },
    swatchTextLight: {
        color: theme.colors.primary.white,
    },
});
