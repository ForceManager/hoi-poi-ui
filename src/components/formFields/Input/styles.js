export default (theme) => {
    const titleBorder = {
        backgroundColor: theme.colors.neutralBase,
        border: '1px solid transparent',
        borderBottom: `1px solid ${theme.colors.neutral400}`,
    };
    return {
        root: {
            outline: 'none',
        },
        title: {
            '& $formControl': {
                padding: 0,
                ...titleBorder,
            },
            '&$focused': {
                '& $formControl': {
                    ...titleBorder,
                },
            },
            '&$error': {
                '& $formControl': {
                    ...titleBorder,
                },
            },
            '&$isReadOnly': {
                '& $formControl': {
                    ...titleBorder,
                },
            },
            '& $input': {
                ...theme.typography.h5,
            },
            '& $info': {
                marginLeft: 0,
            },
            '& $errorInfo': {
                marginLeft: 0,
            },
        },
        Label: {
            width: 153,
            marginRight: 17,
            padding: '0 0 8px 0',
        },
        vertical: {
            display: 'block',
            padding: '15px 0 10px 0',
            marginLeft: '0 !important',
            '& $Label': {
                width: '100%',
            },
        },
        horizontal: {
            display: 'flex',
            alignItems: 'center',
            padding: '15px 0 10px 0',
            '& $error': {},
            '& $Label': {
                padding: 0,
            },
        },
        info: {
            ...theme.typography.caption,
            position: 'absolute',
            marginLeft: -10,
            lineHeight: '20px',
            bottom: -24,
            color: theme.colors.neutral600,
        },
        errorInfo: {
            ...theme.typography.caption,
            position: 'absolute',
            marginLeft: -10,
            lineHeight: '20px',
            bottom: -24,
            color: theme.colors.red500,
        },
        errorInfoSecond: {
            bottom: -42,
        },
        formControl: {
            position: 'relative',
            display: 'flex',
            alignItems: 'center',
            width: 300,
            boxSizing: 'border-box',
            padding: '0 10px',
            lineHeight: 0,
            borderRadius: '4px',
            border: '1px solid transparent',
            backgroundColor: theme.colors.neutral200,
            transition: 'all 0.15s ease',
        },
        input: {
            ...theme.typography.body,
            background: 'transparent',
            color: theme.colors.neutral900,
            width: '100%',
            lineHeight: '20px',
            height: '40px',
            alignItems: 'center',
            boxSizing: 'border-box',
            flexGrow: 1,
            outline: 'none',
            border: 'none',
            '&::placeholder': {
                color: theme.colors.neutral600,
            },
            '&::-webkit-outer-spin-button': {
                '-webkit-appearance': 'none',
                margin: 0,
            },
            '&::-webkit-inner-spin-button': {
                '-webkit-appearance': 'none',
                margin: 0,
            },
            '&[type=number]': {
                '-moz-appearance': 'textfield',
            },
        },
        withMessage: {
            paddingBottom: '30px !important',
        },
        withTwoMessage: {
            paddingBottom: '45px !important',
            '& $error': {
                bottom: -39,
            },
        },
        error: {
            '& $formControl': {
                backgroundColor: theme.colors.neutralBase,
                border: `1px solid ${theme.colors.red500}`,
            },
        },
        isReadOnly: {
            '& $formControl': {
                border: `1px solid ${theme.colors.neutral500}`,
                backgroundColor: theme.colors.neutralBase,
                color: theme.colors.neutral500,
                '&::placeholder': {
                    color: theme.colors.neutral500,
                },
            },
            '& $input': {
                cursor: 'default',
            },
        },
        isReadAndDuplicable: {
            '& $postCloseComponent': {
                display: 'inline-flex',
            },
        },
        isClickable: {
            cursor: 'pointer',
        },
        clear: {
            width: 10,
            height: 10,
            '& span': {
                width: '100%',
                height: '100%',
            },
        },
        copy: {},
        focused: {
            '& $formControl': {
                backgroundColor: theme.colors.neutralBase,
                border: `1px solid ${theme.colors.orange500}`,
            },
        },
        isFullWidth: {
            '& $formControl': {
                width: '100%',
                flex: 1,
            },
            '&$vertical $Label': {
                width: '100%',
            },
        },
        preComponent: {
            marginRight: 5,
        },
        postComponent: {
            marginLeft: 5,
            display: 'flex',
            alignItems: 'center',
        },
        customPostComponent: {
            marginLeft: 5,
        },
        postCloseComponent: {
            '& > span, > div': {
                marginLeft: '5px',
                '&:first-child': {
                    marginLeft: '0px',
                },
            },
        },
        custom: {
            '& $formControl': {
                '&::before': {
                    borderBottom: 'none',
                },
            },
        },
    };
};
