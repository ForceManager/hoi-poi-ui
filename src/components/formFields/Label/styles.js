export default (theme) => ({
    root: {
        ...theme.typography.body,
        lineHeight: '20px',
        color: theme.colors.neutral900,
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'center',
    },
    Text: {
        paddingRight: 5,
    },
    info: {
        display: 'block',
        width: 10,
    },
    icon: {
        display: 'block',
        width: 10,
        height: 10,
        cursor: 'pointer',
        '& path': {
            fill: theme.colors.disabledGrey,
        },
    },
});
