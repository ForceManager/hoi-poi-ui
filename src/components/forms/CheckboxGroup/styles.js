export default (theme) => ({
    root: {},
    Label: {
        width: 150,
        marginRight: 17,
        flexShrink: 0,
    },
    error: {
        ...theme.typography.smallText,
        position: 'absolute',
        lineHeight: '20px',
        bottom: -21,
        color: theme.colors.red500,
    },
    errorPerCheck: {
        ...theme.typography.smallText,
        color: theme.colors.red500,
        margin: '-6px 0 14px',
    },
    formControl: {
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        width: 290,
    },
    isFullWidth: {
        width: '100%',
    },
    checkboxWrapper: {
        display: 'flex',
        alignItems: 'center',
        marginBottom: 14,
        '&:hover $info': {
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
        },
    },
    checkboxControl: {
        display: 'flex',
        alignItems: 'center',
        lineHeight: '20px',
        cursor: 'pointer',
        '&:last-child': {
            marginBottom: 0,
        },
    },
    checkboxLabel: {
        marginLeft: 10,
        ...theme.typography.defaultText,
        color: theme.colors.neutral700,
    },
    iconContainer: {
        height: 18,
        width: 18,
        marginLeft: 5,
    },
    info: {
        display: 'none',
        width: '100%',
        height: '100%',
        cursor: 'pointer',
    },
    popoverTitle: {
        marginBottom: 5,
    },
    isReadOnly: {
        '& $formControl': {
            cursor: 'not-allowed',
            '&::before': {
                borderBottom: `1px solid ${theme.colors.neutral400}`,
            },
            '&::after': {
                borderBottom: `1px solid ${theme.colors.neutral400}`,
            },
        },
    },
    vertical: {
        display: 'block',
        padding: '8px 0',
        marginLeft: '0 !important',
        '& $formControl': {
            marginTop: 10,
        },
    },
    horizontal: {
        display: 'flex',
        alignItems: 'flex-start',
        padding: '8px 0',
    },
});
