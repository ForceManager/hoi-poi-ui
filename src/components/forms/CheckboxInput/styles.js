export default (theme) => ({
    root: {
        outline: 'none',
    },
    Label: {
        width: 153,
        marginRight: 17,
        padding: '0 0 5px 0',
    },
    error: {
        ...theme.typography.smallText,
        position: 'absolute',
        lineHeight: '20px',
        bottom: -21,
        color: theme.colors.red,
    },
    formControl: {
        position: 'relative',
        display: 'flex',
        alignItems: 'center',
        width: 290,
        lineHeight: 0,
    },
    checkbox: {
        width: '100%',
        lineHeight: '20px',
        flexGrow: 1,
        outline: 'none',
        border: 'none',
    },
    errored: {
        paddingBottom: '30px !important',
        '& $formControl': {
            '&::before': {
                borderBottom: `1px solid ${theme.colors.red} !important`,
            },
            '&::after': {
                borderBottom: `1px solid ${theme.colors.red} !important`,
            },
        },
    },
    vertical: {
        display: 'block',
        padding: '15px 0 10px 0',
        marginLeft: '0 !important',
        '& $formControl': {
            marginTop: 10,
        },
        '& $Label': {
            width: '100%',
        },
    },
    horizontal: {
        display: 'flex',
        alignItems: 'center',
        padding: '15px 0 10px 0',
        '& $error': {},
    },
    isFullWidth: {
        '& $formControl': {
            width: '100%',
            flex: 1,
        },
        '&$vertical $Label': {
            width: '100%',
        },
    },
});
