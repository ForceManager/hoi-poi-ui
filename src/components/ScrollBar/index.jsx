import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'react-jss';
import classnames from 'classnames';
import { Scrollbars } from 'react-custom-scrollbars';
import { getOverrides } from '../../utils/overrides';
import styles from './styles';

function ScrollBar({
    children,
    overrides: overridesProp,
    className: classNameProp,
    classes,
    width,
    height,
    autoHide,
    autoHeight,
    autoHeightMin,
    autoHeightMax,
    ...props
}) {
    // Overrides
    const override = getOverrides(overridesProp, ScrollBar.overrides);

    // Classes
    const rootClassName = classnames(classes.root, classNameProp);

    let style = {};
    if (width) style.width = width;
    if (height) style.height = height;

    const rootProps = {
        style,
        autoHide,
        autoHeight,
        autoHeightMin,
        autoHeightMax,
        ...override['react-custom-scrollbars'],
    };

    return (
        <Scrollbars className={rootClassName} {...rootProps}>
            {children}
        </Scrollbars>
    );
}

ScrollBar.defaultProps = {
    autoHide: true,
};

ScrollBar.propTypes = {
    width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    autoHide: PropTypes.bool,
    autoHeight: PropTypes.bool,
    autoHeightMin: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    autoHeightMax: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

export default React.memo(withStyles(styles, { name: 'ScrollBar' })(ScrollBar));
