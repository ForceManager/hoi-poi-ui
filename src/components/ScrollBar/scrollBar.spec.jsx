import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from '../../utils/styles';
import ScrollBar from '../ScrollBar';

describe('ScrollBar', () => {
    test('match snapshot', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <ScrollBar>Lorem ipsum</ScrollBar>
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
