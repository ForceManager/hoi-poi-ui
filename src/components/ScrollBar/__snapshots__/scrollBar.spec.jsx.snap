// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollBar match snapshot 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "danger": "#f04540",
          "dangerDark": "#C43833",
          "grey": "#697782",
          "icons": "#788590",
          "lightBlack": "#5f656b",
          "lightGray": "#f0f3f5",
          "lines": "#dce3eb",
          "overlay": "rgba(33,33,33,0.55)",
          "primary": "#00CED1",
          "primaryDark": "#00a0a0",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "text": Object {
            "danger": "white",
            "default": "#697782",
            "primary": "white",
          },
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(122,122,122,0.23)",
          },
        },
        "typography": Object {
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "lineHeightBase": 1.5,
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi-"
      jss={
        Jss {
          "generateId": [Function],
          "id": 2,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                  "onProcessSheet": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [
                [Function],
              ],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.0.0-alpha.16",
        }
      }
    >
      <JssContextSubscriber>
        <WithStyles(ScrollBar)
          autoHide={true}
          innerRef={null}
          jssContext={
            Object {
              "disableStylesGeneration": false,
              "jss": Jss {
                "generateId": [Function],
                "id": 2,
                "options": Object {
                  "Renderer": [Function],
                  "createGenerateId": [Function],
                  "plugins": Array [],
                },
                "plugins": PluginsRegistry {
                  "plugins": Object {
                    "external": Array [
                      Object {
                        "onCreateRule": [Function],
                        "onProcessStyle": [Function],
                        "onUpdate": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                        "onProcessRule": [Function],
                      },
                      Object {
                        "onProcessRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                        "onProcessRule": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onProcessRule": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onProcessRule": [Function],
                        "onProcessSheet": [Function],
                      },
                    ],
                    "internal": Array [
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onChangeValue": [Function],
                        "onCreateRule": [Function],
                        "onProcessStyle": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                      Object {
                        "onCreateRule": [Function],
                      },
                    ],
                  },
                  "registry": Object {
                    "onChangeValue": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onCreateRule": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onProcessRule": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onProcessSheet": Array [
                      [Function],
                    ],
                    "onProcessStyle": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onUpdate": Array [
                      [Function],
                    ],
                  },
                },
                "version": "10.0.0-alpha.16",
              },
              "managers": Object {
                "0": SheetsManager {
                  "length": 1,
                  "sheets": WeakMap {},
                },
              },
              "sheetOptions": Object {
                "classNamePrefix": "HoiPoi-",
                "generateId": [Function],
              },
            }
          }
          theme={
            Object {
              "colors": Object {
                "danger": "#f04540",
                "dangerDark": "#C43833",
                "grey": "#697782",
                "icons": "#788590",
                "lightBlack": "#5f656b",
                "lightGray": "#f0f3f5",
                "lines": "#dce3eb",
                "overlay": "rgba(33,33,33,0.55)",
                "primary": "#00CED1",
                "primaryDark": "#00a0a0",
                "shadow": "rgba(0, 0, 0, 0.25)",
                "text": Object {
                  "danger": "white",
                  "default": "#697782",
                  "primary": "white",
                },
              },
              "effects": Object {
                "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
                "buttonActive": Object {
                  "boxShadow": "inset 0 -2px 4px 0 rgba(122,122,122,0.23)",
                },
              },
              "typography": Object {
                "button": Object {
                  "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": 11,
                  "lineHeight": 1.5,
                },
                "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSizeBase": 13,
                "lineHeightBase": 1.5,
              },
            }
          }
        >
          <ScrollBar
            autoHide={true}
            classes={Object {}}
          >
            <Scrollbars
              autoHeight={false}
              autoHeightMax={200}
              autoHeightMin={0}
              autoHide={true}
              autoHideDuration={200}
              autoHideTimeout={1000}
              className=""
              hideTracksWhenNotNeeded={false}
              renderThumbHorizontal={[Function]}
              renderThumbVertical={[Function]}
              renderTrackHorizontal={[Function]}
              renderTrackVertical={[Function]}
              renderView={[Function]}
              style={Object {}}
              tagName="div"
              thumbMinSize={30}
              universal={false}
            >
              <div
                className=""
                style={
                  Object {
                    "height": "100%",
                    "overflow": "hidden",
                    "position": "relative",
                    "width": "100%",
                  }
                }
              >
                <div
                  key="view"
                  style={
                    Object {
                      "WebkitOverflowScrolling": "touch",
                      "bottom": 0,
                      "left": 0,
                      "marginBottom": 0,
                      "marginRight": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  Lorem ipsum
                </div>
                <div
                  key="trackHorizontal"
                  style={
                    Object {
                      "borderRadius": 3,
                      "bottom": 2,
                      "display": "none",
                      "height": 6,
                      "left": 2,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 2,
                      "transition": "opacity 200ms",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "backgroundColor": "rgba(0,0,0,.2)",
                        "borderRadius": "inherit",
                        "cursor": "pointer",
                        "display": "block",
                        "height": "100%",
                        "position": "relative",
                      }
                    }
                  />
                </div>
                <div
                  key="trackVertical"
                  style={
                    Object {
                      "borderRadius": 3,
                      "bottom": 2,
                      "display": "none",
                      "opacity": 0,
                      "position": "absolute",
                      "right": 2,
                      "top": 2,
                      "transition": "opacity 200ms",
                      "width": 6,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "backgroundColor": "rgba(0,0,0,.2)",
                        "borderRadius": "inherit",
                        "cursor": "pointer",
                        "display": "block",
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
            </Scrollbars>
          </ScrollBar>
        </WithStyles(ScrollBar)>
      </JssContextSubscriber>
    </JssProvider>
  </ThemeProvider>
</Component>
`;
